presto-parquet/src/test/java/com/facebook/presto/parquet/writer/TestParquetWriter.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.parquet.writer;
+
+import com.facebook.presto.common.PageBuilder;
+import com.facebook.presto.common.type.Type;
+import com.google.common.collect.ImmutableList;
+import io.airlift.units.DataSize;
+import org.apache.parquet.hadoop.metadata.CompressionCodecName;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+
+import static com.facebook.presto.common.type.BigintType.BIGINT;
+import static com.facebook.presto.common.type.BooleanType.BOOLEAN;
+import static com.facebook.presto.common.type.IntegerType.INTEGER;
+import static com.facebook.presto.common.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.io.Files.createTempDir;
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.UUID.randomUUID;
+import static org.testng.Assert.fail;
+
+public class TestParquetWriter
+{
+    private File temporaryDirectory;
+    private File parquetFile;
+
+    @BeforeMethod
+    public void setup()
+    {
+        temporaryDirectory = createTempDir();
+        parquetFile = new File(temporaryDirectory, randomUUID().toString());
+    }
+
+    @AfterMethod
+    public void tearDown()
+            throws IOException
+    {
+        deleteRecursively(temporaryDirectory.toPath(), ALLOW_INSECURE);
+    }
+
+    @Test
+    public void testRowGroupFlushInterleavedColumnWriterFallbacks()
```

Comment:
It looks like as long as no exception is thrown, the test will pass. This PR is to let the max dictionary size take effect. Can you share on this test validate "if I set a value for max directory size, the dictionary page should be no more than that after the parquet is generated"?

Reply:
Sorry that I may not have described it clearly. The main purpose of this PR is not about page size setting, they are introduced mainly for unit test to reappear the problem faster. This PR's main purpose is to fix the bug described above when writing data to parquet. For example, when I try to write data to a parquet formatted iceberg table by executing "create table iceberg.default.store_sales as select * from tpcds.sf10.store_sales", even in default page settings, it will fail and throw an exception seems like 'java.lang.ArrayIndexOutOfBoundsException: 33186'.

Reply:
Sorry that I may not have described it clearly. The main purpose of this PR is not about page size setting, they are introduced mainly for unit test to reappear the problem faster. This PR's main purpose is to fix the bug described above when writing data to parquet. For example, when I try to write data to a parquet formatted iceberg table by executing "create table iceberg.default.store_sales as select * from tpcds.sf10.store_sales", even in default page settings, it will fail and throw an exception seems like 'java.lang.ArrayIndexOutOfBoundsException: 33186'.